### PR TITLE
Linear Regression function implemented

### PR DESCRIPTION
**Pull request by Madhav2310**

### CODE REVIEW

**Comment by Madhav2310** [comment]
@certik  @czgdp1807 @Smit-create Please review

File: integration_tests/test_statistics.py [hunk 1]
```diff
@@ -97,6 +97,18 @@ def test_stdev():
...
 
+def test_linear_regression():
+    c: list[f64]
+    c = [2.74, 1.23, 2.63, 2.22, 3.0, 1.98]
+    d: list[f64]
+    d = [9.4, 1.23, 2.63, 22.4, 1.9, 13.98]
+
+    slope: f64
+    intercept: f64
+    slope, intercept = linear_regression(c,d)
```
Comment (Smit-create) [inline integration_tests/test_statistics.py:108]:
```suggestion
    slope, intercept = linear_regression(c, d)
```

**Review by Smit-create — COMMENTED** [review]
Also add the tests for `linear_regression(list[i32], list[i32])`

**Review by czgdp1807 — COMMENTED** [review]
Looks good so far.

File: integration_tests/test_statistics.py (hunk 1 above)
Comment (Smit-create) [reply integration_tests/test_statistics.py:107]: @Madhav2310

Comment (Madhav2310) [reply integration_tests/test_statistics.py]: Thank you! I'm implemented this change too, please review and merge

Comment (Smit-create) [reply integration_tests/test_statistics.py]: > Also add the tests for linear_regression(list[i32], list[i32])

**Review by Smit-create — APPROVED** [review]

**Pull request merged**